central/resourcecollection/datastore/datastore_impl.go

```diff
@@ -427,3 +431,161 @@ func verifyCollectionObjectNotEmpty(obj *storage.ResourceCollection) error {
 	}
 	return nil
 }
+
+func (ds *datastoreImpl) ResolveListDeployments(ctx context.Context, collection *storage.ResourceCollection) ([]*storage.ListDeployment, error) {
+
+	if err := verifyCollectionConstraints(collection); err != nil {
+		return nil, err
+	}
+
+	// currently we only support one resource selector per collection from UX
+	if collection.GetResourceSelectors() != nil && len(collection.GetResourceSelectors()) > 1 {
+		return nil, errors.New("only 1 resource selector is supported per collection")
+	}
+
+	query, err := ds.resolveCollectionQuery(ctx, collection)
+	if err != nil {
+		return nil, err
+	}
+	return ds.deploymentDS.SearchListDeployments(ctx, query)
+}
+
+func (ds *datastoreImpl) resolveCollectionQuery(ctx context.Context, collection *storage.ResourceCollection) (*v1.Query, error) {
+	var collections []*storage.ResourceCollection
+	var collectionSet set.Set[string]
+	var disjunctions []*v1.Query
+
+	collections = append(collections, collection)
+
+	ds.lock.RLock()
+	defer ds.lock.RUnlock()
+
+	for len(collections) > 0 {
+
+		// get first index and remove from list
+		collection := collections[0]
+		collections = collections[1:]
+
+		if !collectionSet.Add(collection.GetId()) {
+			continue
+		}
+
+		// resolve the collection to queries
+		queries, err := collectionToQueries(collection)
+		if err != nil {
+			return nil, err
+		}
+		disjunctions = append(disjunctions, queries...)
+
+		// add embedded values
+		embeddedList, _, err := ds.storage.GetMany(ctx, embeddedCollectionsToIDList(collection.GetEmbeddedCollections()))
+		if err != nil {
+			return nil, err
+		}
+		collections = append(collections, embeddedList...)
+	}
+
+	return pkgSearch.DisjunctionQuery(disjunctions...), nil
+}
+
+// collectionToQueries returns a list of queries derived from the given resource collection's storage.ResourceSelector list
+// these should be combined as disjunct with any resolved embedded queries
+func collectionToQueries(collection *storage.ResourceCollection) ([]*v1.Query, error) {
```

Comment:
put it under `resourcecollection/utils/`

Reply:
Same comment as below